src/main/java/unicash/logic/Logic.java

```diff
@@ -42,4 +43,9 @@ public interface Logic {
      * Returns an unmodifiable view of the filtered list of transactions
      */
     ObservableList<Transaction> getFilteredTransactionList();
+
+    /**
+     * Returns a HashMap of the expense summary from the model
+     */
+    HashMap<String, Double> getExpenseSummary();
```

Comment:
I thought of this too, initially. It is definitely possible to do the calculations by the caller (i.e. either MainWindow or SummaryWindow). However, I implemented it this way because I thought that calculations should be done by the Model instead. The result will then be passed to Logic, which will then be passed to MainWindow, and then SummaryWindow.

This is simply to prevent the UI components from needing to perform any calculations.

Reply:
Hmm but that thing is, UI components do perform internal calculations such as with the Rolling Balance, but I understand what you mean, and SummaryStatistics is not supposed to be reliant on the TransactionsListPanel like the other UI components, so all in all I think it's ok!

For v1.3b we can consider if we want to change this to a generatic getTransactionTypeSummary, but since we haven't decided if we want a chart for income also, I think this works! Thanks for the clarification!

Reply:
Thinking about it, I think I can actually add another `Static Balance` counter beside Rolling balance which would not depend on the TransactionsListPanel calculation and show the absolute balance at all times, so this might be a benefit!